modules/UI/UI.js

```diff
@@ -1235,17 +1239,11 @@ UI.showDeviceErrorDialog = function (micError, cameraError) {
     }
 
     if (micError) {
-        localStoragePropName += "-mic-" + micError.name;
+        dontShowAgain.localStorageKey += "-mic-" + micError.name;
     }
 
     if (cameraError) {
-        localStoragePropName += "-camera-" + cameraError.name;
-    }
-
-    if (showDoNotShowWarning) {
-        if (window.localStorage[localStoragePropName] === "true") {
-            return;
-        }
+        dontShowAgain.localStorageKey += "-camera-" + cameraError.name;
     }
```

Comment:
Here and above you modify localStorageKey to have -camera... or -mic... but later in dontShowTheDialog you just check for the dontShowAgain.id, but you set in localStorage the localStorageKey. It will not work on second attempt as it have one key set, and other key checked.